web/modules/custom/custom_weather/src/Plugin/Block/CustomWeather.php

```diff
@@ -25,18 +26,13 @@ class CustomWeather extends BlockBase implements ContainerFactoryPluginInterface
    */
   protected string $apiKey;
 
-  /**
-   * Stores the selected city.
-   */
-  protected string $selectedCity;
-
   /**
    * {@inheritdoc}
    */
-  public function __construct(protected ClientFactory $httpClient, array $configuration, $plugin_id, $plugin_definition, protected ConfigFactoryInterface $configFactory) {
+  public function __construct(protected ClientFactory $httpClient, array $configuration, $plugin_id, $plugin_definition, protected ConfigFactoryInterface $configFactory, protected UserCityHandler $city) {
     parent::__construct($configuration, $plugin_id, $plugin_definition);
     $this->apiKey = $this->configFactory->get('custom_weather.settings')->get('api_key');
-    $this->selectedCity = $this->configFactory->get('custom_weather.settings')->get('selected_city');
+    $this->databaseService = $city;
```

Comment:
same, constructor promotion is not that complex to understand
Or create task to learn OOP for a few hours if that could help you